### PR TITLE
tl.dot(a,b, trans_b=True) is not supported by triton2.0+ , updating this api

### PR DESCRIPTION
**Pull request by bmedishe**
Stable Diffusion Inference with deepspeed inference engine works with this update with triton2.1 on A100

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
@jeffdaily @loadams Is there a plan to update DeepSpeed inference extension to use a newer 2.1 triton?

**Comment by loadams** [comment]
> @jeffra @loadams Is there a plan to update DeepSpeed inference extension to use a newer 2.1 triton?

We have it on the backlog and hadn't gotten around to it, will try to prioritize it.

**Comment by loadams** [comment]
I believe we will need to run this manually to test - or that we have a SD unit test PR we can merge this into, I"ll work on that.

File: requirements/requirements-sd.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 diffusers
-triton==2.0.0.dev20221202
+triton
```
[the remark this answers is not in the document]
Comment (loadams) [reply requirements/requirements-sd.txt:2]: Yes we will want to pin this, but for now this is just to test and confirm that things work with 2.1

**Comment by bmedishe** [comment]
> I believe we will need to run this manually to test - or that we have a SD unit test PR we can merge this into, I"ll work on that.

@loadams  could you please review https://github.com/microsoft/DeepSpeed/pull/4573 , it is based on newer https://github.com/ROCmSoftwarePlatform/triton/blob/triton-mlir/python/tutorials/06-fused-attention.py , and does not use tl.trans api

**Review by lekurile — APPROVED** [review]
LGTM!

**Review by mrwyattii — APPROVED** [review]
Thanks @bmedishe

**Pull request merged**